// rsuite/lib/mixins/FormControlMixin.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FormControlMixin = {
    propTypes: {
        onChange: _react2.default.PropTypes.func,
        value: _react2.default.PropTypes.any,
        isValid: _react2.default.PropTypes.bool,
        errorMessage: _react2.default.PropTypes.string,
        onError: _react2.default.PropTypes.func
    },
    shouldCallOnError: function shouldCallOnError() {
        return this.props.isValid === false;
    },
    callError: function callError() {
        var _props = this.props;
        var onError = _props.onError;
        var isValid = _props.isValid;
        var errorMessage = _props.errorMessage;

        onError && onError(!isValid, errorMessage);
    },
    componentDidMount: function componentDidMount() {

        if (this.shouldCallOnError()) {
            this.callError();
        }
    },
    componentDidUpdate: function componentDidUpdate() {
        if (this.shouldCallOnError()) {
            this.callError();
        }
    }
};

exports.default = FormControlMixin;
