// rsuite/lib/BreadcrumbItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Anchor = require('./Anchor');

var _Anchor2 = _interopRequireDefault(_Anchor);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var BreadcrumbItem = _react2.default.createClass({
    displayName: 'BreadcrumbItem',

    propTypes: {
        active: _react2.default.PropTypes.bool,
        /**
         * HTML id for the wrapper `li` element
         */
        id: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.string, _react2.default.PropTypes.number]),
        /**
         * HTML id for the inner `a` element
         */
        linkId: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.string, _react2.default.PropTypes.number]),
        href: _react2.default.PropTypes.string,
        title: _react2.default.PropTypes.node,
        target: _react2.default.PropTypes.string
    },

    getDefaultProps: function getDefaultProps() {
        return {
            active: false
        };
    },
    renderItem: function renderItem(props, linkProps) {

        if (this.props.active) {
            return _react2.default.createElement(
                'span',
                props,
                ' ',
                this.props.children,
                ' '
            );
        }

        return _react2.default.createElement(
            _Anchor2.default,
            _extends({}, props, linkProps),
            this.props.children
        );
    },
    render: function render() {
        var _props = this.props;
        var active = _props.active;
        var className = _props.className;
        var id = _props.id;
        var linkId = _props.linkId;
        var children = _props.children;
        var href = _props.href;
        var title = _props.title;
        var target = _props.target;

        var props = _objectWithoutProperties(_props, ['active', 'className', 'id', 'linkId', 'children', 'href', 'title', 'target']);

        var linkProps = {
            href: href,
            title: title,
            target: target,
            id: linkId
        };

        return _react2.default.createElement(
            'li',
            { id: id, className: (0, _classnames2.default)(className, { active: active }) },
            this.renderItem()
        );
    }
});

exports.default = BreadcrumbItem;
