// rsuite/lib/Button.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _ClassNameMixin = require('./mixins/ClassNameMixin');

var _ClassNameMixin2 = _interopRequireDefault(_ClassNameMixin);

var _Anchor = require('./Anchor');

var _Anchor2 = _interopRequireDefault(_Anchor);

var _elementType = require('./prop-types/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

var Button = _react2.default.createClass({
    displayName: 'Button',

    mixins: [_ClassNameMixin2.default],
    propTypes: {
        active: _react2.default.PropTypes.bool,
        disabled: _react2.default.PropTypes.bool,
        block: _react2.default.PropTypes.bool,
        href: _react2.default.PropTypes.string,
        target: _react2.default.PropTypes.string,
        componentClass: _elementType2.default,
        classPrefix: _react2.default.PropTypes.string,
        type: _react2.default.PropTypes.oneOf(['button', 'reset', 'submit'])
    },
    getDefaultProps: function getDefaultProps() {
        return {
            classPrefix: 'btn',
            active: false,
            type: 'button',
            disabled: false,
            block: false
        };
    },
    renderAnchor: function renderAnchor(classes) {
        var Component = this.props.componentClass || _Anchor2.default;
        var href = this.props.href || '#';

        return _react2.default.createElement(
            Component,
            _extends({}, this.props, {
                href: href,
                className: classes,
                role: 'button' }),
            this.props.children
        );
    },
    renderButton: function renderButton(classes) {
        var Component = this.props.componentClass || 'button';
        return _react2.default.createElement(
            Component,
            _extends({}, this.props, {
                className: classes
            }),
            this.props.children
        );
    },
    render: function render() {
        var classes = _classnames2.default.apply(undefined, [_defineProperty({
            btn: true,
            active: this.props.active,
            disabled: this.props.disabled
        }, this.prefix('block'), this.props.block)].concat(_toConsumableArray(this.getClassNames()), [this.props.className]));

        var renderName = this.props.href || this.props.target ? 'renderAnchor' : 'renderButton';
        return this[renderName](classes);
    }
});

exports.default = Button;
