// rsuite/lib/CheckboxList.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Checkbox = require('./Checkbox');

var _Checkbox2 = _interopRequireDefault(_Checkbox);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CheckboxList = _react2.default.createClass({
    displayName: 'CheckboxList',

    propTypes: {
        name: _react2.default.PropTypes.string,
        inline: _react2.default.PropTypes.bool,
        value: _react2.default.PropTypes.array
    },
    contextTypes: {
        formGroup: _react2.default.PropTypes.object
    },
    getFormGroup: function getFormGroup() {
        return this.context.formGroup || {};
    },
    handleChange: function handleChange(checked) {

        var refs = this.refs;
        var value = [];
        var onChange = this.props.onChange;

        var _getFormGroup = this.getFormGroup();

        var onFormGroupChange = _getFormGroup.onChange;


        setTimeout(function () {
            for (var key in refs) {
                if (refs[key].state.checked) {
                    value.push(refs[key].props.value);
                }
            }
            onChange && onChange(value);
            onFormGroupChange && onFormGroupChange(value);
        }, 1);
    },
    render: function render() {
        var _this = this;

        var _props = this.props;
        var className = _props.className;
        var inline = _props.inline;
        var name = _props.name;
        var children = _props.children;


        var clesses = (0, _classnames2.default)({
            'checkbox-list': true
        }, className);

        var items = _react2.default.Children.map(children, function (child, index) {
            return _react2.default.cloneElement(child, {
                key: index,
                ref: 'checkbox_' + index,
                inline: inline,
                onChange: _this.handleChange,
                name: name
            }, child.props.children);
        });

        return _react2.default.createElement(
            'div',
            {
                className: clesses,
                role: 'checkbox-list'
            },
            items
        );
    }
});

exports.default = CheckboxList;
