// rsuite/lib/Col.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _elementType = require('./prop-types/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SIZES = ['xs', 'sm', 'md', 'lg'];
var NUMBERS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];

var Col = _react2.default.createClass({
    displayName: 'Col',

    propTypes: {

        //class-prefix `col-xs-` `col-sm-` `col-md-`  `col-lg-`
        xs: _react2.default.PropTypes.oneOf(NUMBERS),
        sm: _react2.default.PropTypes.oneOf(NUMBERS),
        md: _react2.default.PropTypes.oneOf(NUMBERS),
        lg: _react2.default.PropTypes.oneOf(NUMBERS),

        //class-prefix `col-xs-offset-` `col-sm-offset-` `col-md-offset-`  `col-lg-offset-`
        xsOffset: _react2.default.PropTypes.oneOf(NUMBERS),
        smOffset: _react2.default.PropTypes.oneOf(NUMBERS),
        mdOffset: _react2.default.PropTypes.oneOf(NUMBERS),
        lgOffset: _react2.default.PropTypes.oneOf(NUMBERS),

        //class-prefix `col-xs-push-` `col-sm-push-` `col-md-push-`  `col-lg-push-`
        xsPush: _react2.default.PropTypes.oneOf(NUMBERS),
        smPush: _react2.default.PropTypes.oneOf(NUMBERS),
        mdPush: _react2.default.PropTypes.oneOf(NUMBERS),
        lgPush: _react2.default.PropTypes.oneOf(NUMBERS),

        //class-prefix `col-xs-pull-` `col-sm-pull-` `col-md-pull-`  `col-lg-pull-`
        xsPull: _react2.default.PropTypes.oneOf(NUMBERS),
        smPull: _react2.default.PropTypes.oneOf(NUMBERS),
        mdPull: _react2.default.PropTypes.oneOf(NUMBERS),
        lgPull: _react2.default.PropTypes.oneOf(NUMBERS),

        //adds class `hidden-xs` `hidden-sm` `hidden-md`  `hidden-lg`
        xsHidden: _react2.default.PropTypes.bool,
        smHidden: _react2.default.PropTypes.bool,
        mdHidden: _react2.default.PropTypes.bool,
        lgHidden: _react2.default.PropTypes.bool,

        componentClass: _elementType2.default
    },

    getDefaultProps: function getDefaultProps() {
        return {
            componentClass: 'div'
        };
    },
    render: function render() {
        var _this = this;

        var Component = this.props.componentClass;
        var classes = {};

        SIZES.map(function (size) {
            var offset = _this.props[size + 'Offset'];
            var push = _this.props[size + 'Push'];
            var pull = _this.props[size + 'Pull'];

            classes['hidden-' + size] = _this.props[size + 'Hidden'];
            classes['col-' + size + '-' + _this.props[size]] = _this.props[size] > 0;
            classes['col-' + size + '-offset-' + offset] = offset >= 0;
            classes['col-' + size + '-push-' + push] = push >= 0;
            classes['col-' + size + '-pull-' + pull] = pull >= 0;
        });

        return _react2.default.createElement(
            Component,
            _extends({}, this.props, { className: (0, _classnames2.default)(this.props.className, classes) }),
            this.props.children
        );
    }
});

exports.default = Col;
