// rsuite/lib/DropdownMenu.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _DropdownMenuItem = require('./DropdownMenuItem');

var _DropdownMenuItem2 = _interopRequireDefault(_DropdownMenuItem);

var _ClassNameMixin = require('./mixins/ClassNameMixin');

var _ClassNameMixin2 = _interopRequireDefault(_ClassNameMixin);

var _createChainedFunction = require('./utils/createChainedFunction');

var _createChainedFunction2 = _interopRequireDefault(_createChainedFunction);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var DorpdownMenu = _react2.default.createClass({
    displayName: 'DorpdownMenu',

    mixins: [_ClassNameMixin2.default],
    propTypes: {
        pullRight: _react2.default.PropTypes.bool,
        onClose: _react2.default.PropTypes.func,
        onSelect: _react2.default.PropTypes.func
    },
    getDefaultProps: function getDefaultProps() {
        return {
            classPrefix: 'dropdown',
            pullRight: false
        };
    },
    getFocusableMenuItems: function getFocusableMenuItems() {
        var menuNode = ReactDOM.findDOMNode(this);
        if (menuNode === undefined) {
            return [];
        }
        return Array.from(menuNode.querySelectorAll('[role="menu-item"]'));
    },
    getItemsAndActiveIndex: function getItemsAndActiveIndex() {
        var items = this.getFocusableMenuItems();
        var activeItemIndex = items.indexOf(document.activeElement);

        return {
            items: items,
            activeItemIndex: activeItemIndex
        };
    },
    handleSelect: function handleSelect(event) {
        var onClose = this.props.onClose;

        onClose && onClose();
    },
    render: function render() {
        var _this = this,
            _classes;

        var _props = this.props;
        var pullRight = _props.pullRight;
        var children = _props.children;
        var className = _props.className;
        var activeKey = _props.activeKey;

        var props = _objectWithoutProperties(_props, ['pullRight', 'children', 'className', 'activeKey']);

        var items = _react2.default.Children.map(children, function (item, index) {
            var childProps = {
                key: index,
                ref: 'menu_item_' + index,
                onSelect: (0, _createChainedFunction2.default)(_this.handleSelect, _this.props.onSelect)
            };

            if (activeKey) {
                childProps.active = activeKey === item.props.eventKey;
            }

            return _react2.default.cloneElement(item, childProps, item.props.children);
        });

        var classes = (_classes = {}, _defineProperty(_classes, this.prefix('menu'), true), _defineProperty(_classes, this.prefix('menu-right'), pullRight), _classes);

        return _react2.default.createElement(
            'ul',
            _extends({}, props, {
                className: (0, _classnames2.default)(className, classes),
                role: 'menu'
            }),
            items
        );
    }
});

exports.default = DorpdownMenu;
