// rsuite/lib/Grid.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _elementType = require('./prop-types/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Grid = _react2.default.createClass({
    displayName: 'Grid',

    propTypes: {
        fluid: _react2.default.PropTypes.bool,
        componentClass: _elementType2.default
    },

    getDefaultProps: function getDefaultProps() {
        return {
            componentClass: 'div',
            fluid: false
        };
    },
    render: function render() {
        var _props = this.props;
        var Component = _props.componentClass;
        var fluid = _props.fluid;
        var className = _props.className;
        var children = _props.children;

        var props = _objectWithoutProperties(_props, ['componentClass', 'fluid', 'className', 'children']);

        var clesses = (0, _classnames2.default)(_defineProperty({}, 'container' + (fluid ? '-fluid' : ''), true), className);

        return _react2.default.createElement(
            Component,
            _extends({}, props, {
                className: clesses }),
            children
        );
    }
});

exports.default = Grid;
