// rsuite/lib/HelpBlock.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HelpBlock = _react2.default.createClass({
    displayName: 'HelpBlock',

    propTypes: {
        htmlFor: _react2.default.PropTypes.string
    },
    contextTypes: {
        formGroup: _react2.default.PropTypes.object
    },
    render: function render() {
        var _context$formGroup = this.context.formGroup;
        var controlId = _context$formGroup.controlId;
        var isValid = _context$formGroup.isValid;
        var errorMessage = _context$formGroup.errorMessage;
        var _props = this.props;
        var className = _props.className;
        var _props$htmlFor = _props.htmlFor;
        var htmlFor = _props$htmlFor === undefined ? controlId : _props$htmlFor;
        var children = _props.children;


        var classes = (0, _classnames2.default)({
            'help-block': true,
            'error': !isValid
        }, className);

        return _react2.default.createElement(
            'span',
            {
                htmlFor: htmlFor,
                className: classes
            },
            !isValid && errorMessage || children
        );
    }
});

exports.default = HelpBlock;
