// rsuite/lib/ModalDialog.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ClassNameMixin = require('./mixins/ClassNameMixin');

var _ClassNameMixin2 = _interopRequireDefault(_ClassNameMixin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var ModalDialog = _react2.default.createClass({
    displayName: 'ModalDialog',

    mixins: [_ClassNameMixin2.default],
    propTypes: {
        /**
         * A css class to apply to the Modal dialog DOM node.
         */
        dialogClassName: _react2.default.PropTypes.string
    },

    render: function render() {
        var _props = this.props;
        var className = _props.className;
        var style = _props.style;

        var props = _objectWithoutProperties(_props, ['className', 'style']);

        var modalStyle = _extends({
            display: 'block'
        }, style);
        var modalClasses = (0, _classnames2.default)('modal', className);
        var dialogClasses = _classnames2.default.apply(undefined, [this.prefix('dialog'), this.props.dialogClassName].concat(_toConsumableArray(this.getClassNames())));

        return _react2.default.createElement(
            'div',
            _extends({
                title: null,
                tabIndex: '-1',
                role: 'dialog',
                style: modalStyle,
                className: modalClasses
            }, props),
            _react2.default.createElement(
                'div',
                { className: dialogClasses },
                _react2.default.createElement(
                    'div',
                    { className: this.prefix('content'), role: 'document' },
                    this.props.children
                )
            )
        );
    }
});

exports.default = ModalDialog;
