// rsuite/lib/NavItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _elementType = require('./prop-types/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

var _createChainedFunction = require('./utils/createChainedFunction');

var _createChainedFunction2 = _interopRequireDefault(_createChainedFunction);

var _Anchor = require('./Anchor');

var _Anchor2 = _interopRequireDefault(_Anchor);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var NavItem = _react2.default.createClass({
    displayName: 'NavItem',


    propTypes: {
        active: _react2.default.PropTypes.bool,
        disabled: _react2.default.PropTypes.bool,
        href: _react2.default.PropTypes.string,
        onClick: _react2.default.PropTypes.func,
        onSelect: _react2.default.PropTypes.func,
        eventKey: _react2.default.PropTypes.any,
        componentClass: _elementType2.default
    },

    getDefaultProps: function getDefaultProps() {
        return {
            href: '',
            active: false,
            disabled: false
        };
    },
    handleClick: function handleClick(e) {

        if (this.props.onSelect) {
            if (!this.props.disabled) {
                this.props.onSelect(this.props.eventKey, e);
            }
        }
    },
    render: function render() {
        var _props = this.props;
        var active = _props.active;
        var disabled = _props.disabled;
        var role = _props.role;
        var href = _props.href;
        var onClick = _props.onClick;
        var className = _props.className;
        var style = _props.style;
        var onSelect = _props.onSelect;
        var eventKey = _props.eventKey;
        var children = _props.children;

        var props = _objectWithoutProperties(_props, ['active', 'disabled', 'role', 'href', 'onClick', 'className', 'style', 'onSelect', 'eventKey', 'children']);

        var classes = (0, _classnames2.default)({
            active: active,
            disabled: disabled
        }, className);

        var Component = this.props.componentClass || _Anchor2.default;

        return _react2.default.createElement(
            'li',
            { role: 'presentation', className: classes, style: style },
            _react2.default.createElement(
                Component,
                _extends({}, props, {
                    disabled: disabled,
                    role: role,
                    href: href,
                    onClick: (0, _createChainedFunction2.default)(onClick, this.handleClick)
                }),
                children
            )
        );
    }
});

exports.default = NavItem;
