// rsuite/lib/PaginationButton.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _elementType = require('./prop-types/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

var _createChainedFunction = require('./utils/createChainedFunction');

var _createChainedFunction2 = _interopRequireDefault(_createChainedFunction);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var PaginationButton = _react2.default.createClass({
    displayName: 'PaginationButton',


    propTypes: {
        className: _react2.default.PropTypes.string,
        eventKey: _react2.default.PropTypes.any,
        onSelect: _react2.default.PropTypes.func,
        disabled: _react2.default.PropTypes.bool,
        active: _react2.default.PropTypes.bool,
        onClick: _react2.default.PropTypes.func,
        componentClass: _elementType2.default
    },

    getDefaultProps: function getDefaultProps() {
        return {
            active: false,
            disabled: false
        };
    },
    handleClick: function handleClick(event) {
        if (this.props.disabled) {
            return;
        }

        if (this.props.onSelect) {
            this.props.onSelect(this.props.eventKey, event);
        }
    },
    render: function render() {
        var _props = this.props;
        var active = _props.active;
        var disabled = _props.disabled;
        var onClick = _props.onClick;
        var Component = _props.componentClass;
        var className = _props.className;
        var style = _props.style;
        var onSelect = _props.onSelect;

        var props = _objectWithoutProperties(_props, ['active', 'disabled', 'onClick', 'componentClass', 'className', 'style', 'onSelect']);

        return _react2.default.createElement(
            'li',
            {
                className: (0, _classnames2.default)(className, { active: active, disabled: disabled }),
                style: style
            },
            _react2.default.createElement(Component, _extends({}, props, {
                disabled: disabled,
                onClick: (0, _createChainedFunction2.default)(onClick, this.handleClick)
            }))
        );
    }
});

exports.default = PaginationButton;
