// rsuite/lib/Popover.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _ClassNameMixin = require('./mixins/ClassNameMixin');

var _ClassNameMixin2 = _interopRequireDefault(_ClassNameMixin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Popover = _react2.default.createClass({
    displayName: 'Popover',

    mixins: [_ClassNameMixin2.default],
    propTypes: {
        placement: _react2.default.PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
        classPrefix: _react2.default.PropTypes.string,
        title: _react2.default.PropTypes.node
    },
    getDefaultProps: function getDefaultProps() {
        return {
            classPrefix: 'popover',
            placement: 'right'
        };
    },
    render: function render() {
        var classes = (0, _classnames2.default)(_defineProperty({
            popover: true
        }, this.props.placement, true), this.props.className);

        var styles = _extends({
            display: 'block'
        }, this.props.style);

        return _react2.default.createElement(
            'div',
            _extends({ role: 'popover' }, this.props, { className: classes, style: styles }),
            _react2.default.createElement('div', { className: 'arrow' }),
            _react2.default.createElement(
                'h3',
                { className: this.prefix('title') },
                this.props.title
            ),
            _react2.default.createElement(
                'div',
                { className: this.prefix('content') },
                this.props.children
            )
        );
    }
});

exports.default = Popover;
