// rsuite/lib/Sidebar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Sidebar = _react2.default.createClass({
    displayName: 'Sidebar',

    propTypes: {
        pullRight: _react2.default.PropTypes.bool
    },
    contextTypes: {
        page: _react2.default.PropTypes.bool
    },
    render: function render() {
        var _props = this.props;
        var className = _props.className;
        var pullRight = _props.pullRight;

        var props = _objectWithoutProperties(_props, ['className', 'pullRight']);

        var activeClass = this.context.page ? 'page-sidebar' : 'sidebar';
        var wrapperClass = activeClass + '-wrapper';
        var classes = (0, _classnames2.default)(_defineProperty({
            'collapse': true,
            'navbar-collapse': true,
            'right': pullRight
        }, activeClass, true), className);

        return _react2.default.createElement(
            'div',
            _extends({}, props, { className: wrapperClass }),
            _react2.default.createElement(
                'div',
                { className: classes },
                this.props.children
            )
        );
    }
});

exports.default = Sidebar;
