// rsuite/lib/Table.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var Table = _react2.default.createClass({
    displayName: 'Table',

    propTypes: {
        striped: _react2.default.PropTypes.bool,
        bordered: _react2.default.PropTypes.bool,
        condensed: _react2.default.PropTypes.bool,
        hover: _react2.default.PropTypes.bool,
        responsive: _react2.default.PropTypes.bool
    },

    getDefaultProps: function getDefaultProps() {
        return {
            bordered: false,
            condensed: false,
            hover: false,
            responsive: false,
            striped: false
        };
    },
    render: function render() {
        var _props = this.props;
        var striped = _props.striped;
        var bordered = _props.bordered;
        var condensed = _props.condensed;
        var hover = _props.hover;
        var children = _props.children;
        var className = _props.className;

        var props = _objectWithoutProperties(_props, ['striped', 'bordered', 'condensed', 'hover', 'children', 'className']);

        var classes = (0, _classnames2.default)({
            'table': true,
            'table-striped': striped,
            'table-bordered': bordered,
            'table-condensed': condensed,
            'table-hover': hover
        }, className);

        var table = _react2.default.createElement(
            'table',
            _extends({}, props, { className: classes }),
            children
        );

        return this.props.responsive ? _react2.default.createElement(
            'div',
            { className: 'table-responsive' },
            table
        ) : table;
    }
});

exports.default = Table;
