// rsuite/lib/Text.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _elementType = require('./prop-types/elementType');

var _elementType2 = _interopRequireDefault(_elementType);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var SHAPES = ['default', 'muted', 'primary', 'success', 'warning', 'danger', 'info'];

var Text = _react2.default.createClass({
    displayName: 'Text',

    propTypes: {
        bg: _react2.default.PropTypes.bool,
        shape: _react2.default.PropTypes.oneOf(SHAPES),
        componentClass: _elementType2.default
    },
    getDefaultProps: function getDefaultProps() {
        return {
            shape: 'default',
            componentClass: 'p'
        };
    },
    render: function render() {
        var _props = this.props;
        var Component = _props.componentClass;
        var bg = _props.bg;
        var shape = _props.shape;
        var className = _props.className;

        var props = _objectWithoutProperties(_props, ['componentClass', 'bg', 'shape', 'className']);

        var classes = (0, _classnames2.default)(_defineProperty({}, (bg ? 'bg' : 'text') + '-' + shape, true), className);

        return _react2.default.createElement(Component, _extends({}, props, { className: classes }));
    }
});

exports.default = Text;
