// rsuite/lib/Tooltip.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _ClassNameMixin = require('./mixins/ClassNameMixin');

var _ClassNameMixin2 = _interopRequireDefault(_ClassNameMixin);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Tooltip = _react2.default.createClass({
    displayName: 'Tooltip',

    mixins: [_ClassNameMixin2.default],
    propTypes: {
        placement: _react2.default.PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
        positionLeft: _react2.default.PropTypes.number,
        positionTop: _react2.default.PropTypes.number,
        classPrefix: _react2.default.PropTypes.string,
        arrowOffsetLeft: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.number, _react2.default.PropTypes.string]),
        arrowOffsetTop: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.number, _react2.default.PropTypes.string]),
        title: _react2.default.PropTypes.node
    },
    getDefaultProps: function getDefaultProps() {
        return {
            placement: 'right',
            classPrefix: 'tooltip'
        };
    },
    render: function render() {
        var _props = this.props;
        var placement = _props.placement;
        var className = _props.className;
        var positionLeft = _props.positionLeft;
        var positionTop = _props.positionTop;
        var children = _props.children;

        var classes = (0, _classnames2.default)(_defineProperty({
            'tooltip': true
        }, placement, true), className);

        var style = {
            left: positionLeft,
            top: positionTop
        };

        var arrowStyle = {
            left: this.props.arrowOffsetLeft,
            top: this.props.arrowOffsetTop
        };
        return _react2.default.createElement(
            'div',
            _extends({
                role: 'tooltip'
            }, this.props, {
                className: classes,
                style: style
            }),
            _react2.default.createElement('div', { className: this.prefix('arrow'), style: arrowStyle }),
            _react2.default.createElement(
                'div',
                { className: this.prefix('inner') },
                children
            )
        );
    }
});

exports.default = Tooltip;
